File: Q6 best time to sell and buy.py
class Solution:
    def maxProfitFinder(self, prices, i, minPrice, maxProfit):
        if i == len(prices):
            return maxProfit
        if prices[i] < minPrice:
            minPrice = prices[i]
        tProfit = prices[i] - minPrice
        if tProfit > maxProfit:
            maxProfit = tProfit
        return self.maxProfitFinder(prices, i + 1, minPrice, maxProfit)

    def maxProfit(self, prices):
        minPrice = float('inf')
        maxProfit = float('-inf')
        return self.maxProfitFinder(prices, 0, minPrice, maxProfit)

File: Q6 best time to sell and buy.js
var maxProfit = function(prices) {
    function maxProfitFinder(prices, i, minPrice, maxProfit) {
        if (i === prices.length) {
            return maxProfit;
        }
        
        // Update minPrice and maxProfit
        if (prices[i] < minPrice) {
            minPrice = prices[i];
        }
        const tProfit = prices[i] - minPrice;
        if (tProfit > maxProfit) {
            maxProfit = tProfit;
        }
        
        // Recursive call to the next index
        return maxProfitFinder(prices, i + 1, minPrice, maxProfit);
    }
    const minPrice = Infinity;
    const maxProfit = 0;
    return maxProfitFinder(prices, 0, minPrice, maxProfit);
};



